lib/Dialect/Handshake/HandshakeOps.cpp

```diff
@@ -343,8 +343,9 @@ struct EliminateCBranchIntoMuxPattern : OpRewritePattern<MuxOp> {
       rewriter.replaceOp(op, firstParentCBranch.getDataOperand());
     });
 
-    // Remove the cbranch
-    rewriter.eraseOp(firstParentCBranch);
+    // Remove the cbranch if it has no uses
+    if (firstParentCBranch.use_empty())
```

Comment:
That should not even be required as the operation is marked as `Pure`. If the operation has no further uses, it can be safely deleted by the canonicalizer/DCE passes.

Reply:
Did not think of that, thanks! I removed the `eraseOp` call altogether (as you suggest, output is the same in unit tests).